### PR TITLE
Improvements to variant normalization edge cases. Add persistent redis caching.

### PR DESCRIPTION
**Pull request by theferrit32**

### CODE REVIEW

File: src/genegraph/source/registry/core.clj [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+
+#_(defmethod get-key org.rocksdb.RocksDB [db k]
+    (let [{:keys [connection key-serializer]} db]
+    ;; (rocksdb/rocks-get-multipart)
+      (comment
+      ;;Do the implementation
+        )))
+
+(defmethod get-key 'RedisClient [db k]
```
Comment (toneillbroad) [inline src/genegraph/source/registry/core.clj:48]: Shouldn't this be ':RedisClient' aka keyword?

File: src/genegraph/source/registry/redis.clj [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+;; Might be able to get the handle to the threadpool by calling
+;; taoensso.carmine.connections/conn-pool.
+;; Might also be able to kick the threadpool out of hte memoize cache
+;; by sending the db spec with :mem/del as the first vararg.
+;; http://ptaoussanis.github.io/encore/taoensso.encore.html#var-memoize
+(def conn {:pool {} :spec {:uri "redis://localhost:6379/"}})
+
+(defmacro wcar* [& body] `(car/wcar conn ~@body))
+
+(wcar* (car/ping)
```
Comment (toneillbroad) [inline src/genegraph/source/registry/redis.clj:20]: Going to need some explanations here

File: src/genegraph/source/registry/core.clj (hunk 1 above)
Comment (theferrit32) [reply src/genegraph/source/registry/core.clj]: Yeah I was just brainstorming here of some multimethods if we get to the point where we want the k/v store to be more swappable between redis and rocksdb (or others backends). I've removed this file for now.

File: src/genegraph/source/registry/redis.clj (hunk 2 above)
Comment (theferrit32) [reply src/genegraph/source/registry/redis.clj]: `wcar` is a function in the carmine library that handles connection pooling and command pipelining/transactions. So you can pass multiple `car/get` or `car/set` commands in a single `car/wcar` call, and they will be sent in a batch and executed in sequence and the results will all be transmitted back to client in one response. 

It implements this concept:
https://redis.io/docs/manual/pipelining/#redis-pipelining

**Comment by theferrit32** [comment]
The changes made to this PR since last dev meeting was mostly incorporating code feedback from that.
- removed dead code files
- add a maximum to the initial time it will wait for redis to be available.
- add a wait on individual put/get calls to redis to handle the case where the redis pod is recycled 

The last feature was tested by deploying both pods, starting the clinvar-raw ingest (which involved genegraph read/write to redis), and then killing the redis pod in the middle of this, and confirming that genegraph's clinvar-raw processing paused until the redis pod because available again.

The other unrelated feature added was the refactor to create `genegraph.main` and `genegraph.main-repl`.

**Review by toneillbroad — APPROVED** [review]

**Pull request merged**